### PR TITLE
SAAS-880: New New Login Page

### PR DESCRIPTION
**Pull request by TomasBlazek3PG**
Change of design of login page (again).

image

### CODE REVIEW

File: src/core/messages.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
 export const Messages = {
   productDescription:
     'A unified experience for developers and database administrators to monitor, manage, secure, and optimize database environments on any infrastructure.',
+  signIn: 'Sign in to Percona Platform', 
+  workEmail: 'Work Email',
+  linkToSignUp: 'Create one',
```
Comment (fabio-silva) [inline src/core/messages.ts:6]:
```suggestion
  linkToSignUp: 'Create One',
```

File: cypress/integration/auth/logout.spec.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { loginButton } from 'pages/auth/selectors';
-import { dropdownMenu, logoutButton, profileIcon } from 'pages/main/selectors';
 import { getUser } from 'pages/auth/getUser';
+import { commonPage } from 'pages/common.page';
+import signInPage from 'pages/auth/signIn.page';
```
Comment (fabio-silva) [inline cypress/integration/auth/logout.spec.js:3]: Both files follow the same pattern, but one does a default export and the other a named export. Let's stick to one.

File: src/assets/percona-platform-logo.svg [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<svg width="365" height="51" viewBox="0 0 365 51" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (fabio-silva) [inline src/assets/percona-platform-logo.svg:1]: Missing indentation for SVGs

**Review by fabio-silva — CHANGES_REQUESTED** [review]
Please, also check failures from GH's code scanning.
Other than that, this new login page is 🔥

File: src/core/messages.ts (hunk 1 above)
Comment (TomasBlazek3PG) [reply src/core/messages.ts:6]: Are you sure? It looked this way better next to the sentence.

File: cypress/integration/auth/logout.spec.js (hunk 2 above)
Comment (TomasBlazek3PG) [reply cypress/integration/auth/logout.spec.js:3]: Oh this and the scan is part of cypress file problems. @peterSirotnak will surely look at it.

File: src/assets/percona-platform-logo.svg (hunk 3 above)
Comment (TomasBlazek3PG) [reply src/assets/percona-platform-logo.svg:1]: Hmm dont know why figma exports it that way. Will fix it! :)

File: src/core/messages.ts (hunk 1 above)
Comment (fabio-silva) [reply src/core/messages.ts:6]: Oh, it comes after another sentence? Then disregard my comment, didn't realize it, thought it was a single sentence hanging there.

File: cypress/pages/auth/helpers/signIn.helper.ts [hunk 4]
```diff
@@ -1,8 +1,15 @@
+import { timeouts } from '../../../fixtures/timeouts';
 import signInPage from '../signIn.page';
 
 export const fillOutSignInUserDetails = (email?: string, password?: string) => {
   if(email) cy.get(signInPage.locators.emailInput).focus().click().type(email);
 
   cy.get('#idp-discovery-submit').click();
-  if(password) cy.get(signInPage.locators.passwordInput).focus().click().type(password);
+  if(password) cy.get(signInPage.locators.passwordInput, { timeout: timeouts.HALF_MIN })
```
Comment (fabio-silva) [inline cypress/pages/auth/helpers/signIn.helper.ts:8]: we forgot the brackets

File: src/assets/bullet-point-arrow.svg [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z" fill="#E97E30"/>
```
Comment (fabio-silva) [inline src/assets/bullet-point-arrow.svg:2]: we missed this one

File: src/pages/Login/Login.tsx [hunk 6]
```diff
@@ -1,63 +1,57 @@
...
+    oktaAuth.handleLoginRedirect(tokens);
+  }, [oktaAuth]);
+  const onError = useCallback((err) => {
+    console.error('error logging in', err);
+  }, []);
 
-  const doLogin = useCallback(() => {
-    dispatch(loginAction());
-  }, [dispatch]);
+  if (!authState) return null;
```
Comment (fabio-silva) [inline src/pages/Login/Login.tsx:22]: and here as well

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: cypress/integration/auth/logout.spec.js (hunk 2 above)
Comment (fabio-silva) [reply cypress/integration/auth/logout.spec.js:3]: 👍

File: cypress/pages/auth/helpers/signIn.helper.ts (hunk 4 above)
Comment (nicolalamacchia) [inline cypress/pages/auth/helpers/signIn.helper.ts:8]: please, follow the practices that we're following everywhere else. leave a space after the if and always use brackets

File: cypress/pages/helpers/commonPage.helper.ts [hunk 7]
```diff
@@ -4,6 +4,16 @@ export const leftMainMenuClick = (target: LeftMainMenuLinks) => {
...
 };
 
+export const commonPageLoaded = () => {
+  cy.findByTestId(commonPage.locators.perconaLogo);
+  cy.findByTestId(commonPage.locators.profileToggle);
+};
+
+export const uiLogoutUser = () => {
+    cy.findByTestId(commonPage.locators.profileToggle).click();
+    cy.findByTestId(commonPage.locators.logoutButton).click();
```
Comment (nicolalamacchia) [inline cypress/pages/helpers/commonPage.helper.ts:14]: bad indentation (4 spaces instead of 2)

**Review by fabio-silva — APPROVED** [review]

File: src/assets/bullet-point-arrow.svg [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+<path fill-rule="evenodd" clip-rule="evenodd" d="M24.6146 20.0026L16.4596 12.3105L15.3839 13.3259L22.4602 20.0026L15.3839 26.6794L16.4596 27.695L24.6146 20.0026Z" fill="#0A0721"/>
```
Comment (nicolalamacchia) [inline src/assets/bullet-point-arrow.svg]: these need to be indented

File: src/pages/Login/Login.styles.ts [hunk 9]
```diff
@@ -1,47 +1,137 @@
...
+
+      #okta-sign-in.auth-container {
+        border: 0 !important;
+        box-shadow: none !important;
+        background: none !important;
+        margin: 0 !important;
+        box-shadow: none;
+
+        .button-primary {           
+            background: #4C15A5;
```
Comment (nicolalamacchia) [inline src/pages/Login/Login.styles.ts:103]: 4-space indentation mixed with 2-space indentation

File: src/pages/Login/Login.tsx [hunk 10]
```diff
@@ -1,63 +1,57 @@
...
 export const LoginPage: FC = () => {
   const styles = useStyles(getStyles);
-  const dispatch = useDispatch();
+  const { oktaAuth, authState } = useOktaAuth();
+
+  const onSuccess = useCallback((tokens) => {
+    oktaAuth.handleLoginRedirect(tokens);
+  }, [oktaAuth]);
+  const onError = useCallback((err) => {
+    console.error('error logging in', err);
```
Comment (nicolalamacchia) [inline src/pages/Login/Login.tsx:19]: just log `err` here and please use `logger` instead of `console`

File: src/pages/Login/Login.tsx [hunk 11]
```diff
@@ -1,63 +1,57 @@
...
+        <div className={styles.texts}>
+          <h2>
+            {Messages.title}
+          </h2>
+          <h3>
+            {Messages.subtitle}
+          </h3>
+          <ul>
+            {Messages.accessTo.map((point) => (
+              <li>
```
Comment (nicolalamacchia) [inline src/pages/Login/Login.tsx:41]: missing `key` attribute

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 12]
```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useRef } from 'react';
+// @ts-ignore
+import OktaSignIn from '@okta/okta-signin-widget';
+import { useStyles } from '@grafana/ui';
+import { getStyles } from 'pages/Login/Login.styles';
+import { PRIVACY_PMM_URL, TERMS_OF_SERVICE_URL } from 'core/constants';
+import { Messages } from './OktaSignInWidget.messages';
+import '@okta/okta-signin-widget/dist/css/okta-sign-in.min.css';
+
+const insertToS = (context: any) => {
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:10]: we said we'd change it to be more declarative in this iteration. if we can't make it more react-ish for now, at least let's follow some standards, like removing the `const d = document` etc. and open another ticket for that.

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+    const d = document;
+  
+    if (context.controller !== 'registration' || d.getElementsByClassName('tos-label')[0]) {
+      return;
+    }
+  
+    const fieldset = d.getElementsByClassName('o-form-fieldset-container')[0];
+    const requiredLabel = d.getElementsByClassName('required-fields-label')[0];
+
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:19]: mixed 4-space indentation and 2-space indentation (should be 2 spaces)

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: cypress/pages/auth/helpers/signIn.helper.ts (hunk 4 above)
Comment (TomasBlazek3PG) [reply cypress/pages/auth/helpers/signIn.helper.ts:8]: This is not my code. Already tagged @peterSirotnak on it on Nicolas comment.

File: cypress/pages/helpers/commonPage.helper.ts (hunk 7 above)
Comment (TomasBlazek3PG) [reply cypress/pages/helpers/commonPage.helper.ts:14]: Not my code again. I rather let Peter fix it. I can if you insist however.

File: src/assets/bullet-point-arrow.svg (hunk 8 above)
Comment (TomasBlazek3PG) [reply src/assets/bullet-point-arrow.svg]: This one is fixed. Weird it doesnt show you.

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+            <label htmlFor="input099" data-se-for-name="tos" className={cx('tos-label', { checked })}>
+              {Messages.iHaveAgreed}
+              <a href={TERMS_OF_SERVICE_URL} target="_blank" rel="noreferrer">
+                {Messages.tos}
+              </a>
+              {Messages.and}
+              <a href={PRIVACY_PMM_URL} target="_blank" rel="noreferrer">
+                {Messages.privacyPolicy}
+              </a>
+              &nbsp;*
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:63]: nit: please don't use `nbsp`, use CSS instead

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+import { getStyles } from 'pages/Login/Login.styles';
+import { cx } from 'emotion';
+import ReactDOM from 'react-dom';
+import { PRIVACY_PMM_URL, TERMS_OF_SERVICE_URL } from 'core/constants';
+import { Messages } from './OktaSignInWidget.messages';
+import '@okta/okta-signin-widget/dist/css/okta-sign-in.min.css';
+
+const disabledCls = 'link-button-disabled';
+
+interface IToSCheckbox {
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:14]: this should be `ToSCheckboxProps`, as we're doing for all the other components

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+              &nbsp;*
+            </label>
+          </div>
+        </span>
+      </div>
+    </div>
+  );
+};
+
+const insertToS = (context: any, widgetRef: HTMLDivElement) => {
```
Comment (nicolalamacchia) [inline src/components/OktaSignInWidget/OktaSignInWidget.tsx:72]: as per documentation (https://github.com/okta/okta-signin-widget#afterrender), `context` should only contain a `controller` property, which is a string, so we can avoid that `any` here.

File: src/pages/Login/Login.tsx [hunk 17]
```diff
@@ -1,63 +1,59 @@
...
 import { getStyles } from './Login.styles';
+import { Messages } from './Login.messages';
 
 export const LoginPage: FC = () => {
   const styles = useStyles(getStyles);
-  const dispatch = useDispatch();
+  const { oktaAuth, authState } = useOktaAuth();
+
+  const onSuccess = useCallback(
+    (tokens) => {
```
Comment (nicolalamacchia) [inline src/pages/Login/Login.tsx:17]:
```suggestion
  const onSuccess = useCallback((tokens) => {
```

this looks more consistent to what you did with `onError` and what we're doing elsewhere

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.tsx (hunk 17 above)
Comment (TomasBlazek3PG) [reply src/pages/Login/Login.tsx:17]: Oh the prettier is not allowing me to have it this way unfortunatelly

File: src/components/OktaSignInWidget/OktaSignInWidget.tsx (hunk 14 above)
Comment (TomasBlazek3PG) [reply src/components/OktaSignInWidget/OktaSignInWidget.tsx:63]: But here it makes sense. I dont want to break the line and be on the line alone.

**Review by fabio-silva — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**